wagtail/admin/views/pages/edit.py

```diff
@@ -334,6 +334,10 @@ def dispatch(self, request, page_id):
         self.parent = self.page.get_parent()
 
         self.page_perms = self.page.permissions_for_user(self.request.user)
+        self.lock = self.page.get_lock()
+        self.locked_for_user = self.lock is not None and self.lock.for_user(
+            self.request.user
+        )
```

Comment:
And after the change in `page_locked()`, we might be able to change this to 

```suggestion
        self.locked_for_user = self.page_perms.page_locked()
```